src/components/Card/AddressHashCard.tsx

```diff
@@ -69,27 +70,39 @@ const AddressHashCardPanel = styled.div`
       }
     }
   }
-`
 
-const handleHashText = (hash: string) => {
-  if (isSmallMobile()) {
-    return startEndEllipsis(hash, 8, 10)
-  }
-  if (isMediumMobile()) {
-    return startEndEllipsis(hash, 9)
-  }
-  if (isLargeMobile()) {
-    return startEndEllipsis(hash, 14)
+  #address_hash__hash {
+    color: #ffffff;
   }
-  return hash
-}
+`
 
 export default ({ title, hash, dispatch }: { title: string; hash: string; dispatch: AppDispatch }) => {
+  const [hashText, setHashText] = useState(hash)
+  const isMobileDeivce = isMobile()
+
+  useLayoutEffect(() => {
+    if (!isMobile()) {
+      setHashText(hash)
+      return
+    }
+    const contentElement = document.getElementById('address_hash_content')
+    const hashElement = document.getElementById('address_hash__value')
+    if (hashElement && contentElement) {
+      const contentReact = contentElement.getBoundingClientRect()
+      const hashReact = hashElement.getBoundingClientRect()
+      const textWidth = contentReact.width - hashReact.left - 16 - 20
+      const textLength = Math.round(textWidth / 8.0)
+      const startLength = Math.round(textLength / 2)
+      const text = startEndEllipsis(hash, textLength - startLength, startLength)
+      setHashText(text)
+    }
+  }, [hash, isMobileDeivce])
```

Comment:
`useLayoutEffect` doesn't use `isMobileDevice`, so why set ` isMobileDeivce` to dependece ?

Reply:
`useLayoutEffect` maybe use a function will be more better and function name can explain intention.

Reply:
It is not necessary in a production environment.
But if you don't add it, changing the device type will not update the text length when using the chrom test.

Reply:
I've dealt with text length in functions. https://github.com/nervosnetwork/ckb-explorer-frontend/pull/168/commits/282f2e68e4f211d13a2daa74876867310c5cfa35

Reply:
I mean you can move `const isMobileDevice = isMobile()` to `useLayoutEffect`.